### PR TITLE
Feature mvp observable

### PR DESCRIPTION
**Pull request by gaimanjing**
Implement mvp obsevable.
Controller observer view event,
Presenter observe model state.

### CODE REVIEW

File: src/Controller/Controller.hpp [hunk 1]
```diff
@@ -17,25 +24,24 @@ class Controller
...
   bool start(const std::string& filePath, const char* designDocSchemaFilePath = nullptr);
   bool start(const std::vector<char>& buffer, const char* designDocSchemaFilePath = nullptr);
 
-  void onClick(const std::string& path);
-
 private:
-  JsonDocumentObserverPtr m_design_doc_observer;
-  JsonDocumentObserverPtr m_layout_doc_observer;
+  std::shared_ptr<RunLoop> m_run_loop;
+  rxcpp::observer<ModelEventPtr> m_design_doc_observer;
```
Comment (harry75369) [inline src/Controller/Controller.hpp:35]: These observers should be reference of those of Presenter. The reference relationship should be made more clear.

File: src/PlatformAdapter/Native/Exec/NativeExecImpl.cpp [hunk 2]
```diff
@@ -223,6 +216,15 @@ void NativeExecImpl::erase_task(NativeEvalTask* task)
   m_tasks.erase(task);
 }
 
+void NativeExecImpl::stop_node()
```
Comment (harry75369) [inline src/PlatformAdapter/Native/Exec/NativeExecImpl.cpp:219]: better naming? should distinguish from `stop_node_thread_safe`

Comment (gaimanjing) [reply src/PlatformAdapter/Native/Exec/NativeExecImpl.cpp:219]: updated

File: src/Controller/Controller.hpp (hunk 1 above)
Comment (gaimanjing) [reply src/Controller/Controller.hpp:35]: fixed

File: src/Model/ModelEvent.hpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  Invalid,
+  Add,
+  Delete,
+  Update
+};
+
+struct ModelEvent
+{
+  ModelEventType type;
+  std::any data;
```
[the remark this answers is not in the document]
Comment (gaimanjing) [reply src/Model/ModelEvent.hpp:21]: fixed

**Pull request merged**